PyInstaller/depend/analysis.py

```diff
@@ -100,6 +105,43 @@ def __init__(self, pyi_homepath, user_hook_dirs=None, *args, **kwargs):
             os.path.join(self._homepath, 'PyInstaller', 'loader', 'rthooks.dat')
         )
 
+    def msg(self, *args, **kwargs):
+        self._msg(*args, **kwargs)
+
+    def _msg(self, lvl, s, *args):
+        """
+        Print a debug message with the given level
+        """
+        sinfo = None
+        level = None
+
+        # Note: these levels are completely arbitrary and may be adjusted if needed.
+        if s:
+            msg = "%s %s" % (s, ' '.join(map(repr, args)))
+            if lvl <= 1:
+                level = WARNING
+            elif lvl <= 2:
+                level = INFO
+            elif lvl <= 4:
+                level = DEBUG
+
+        if level is None:
+            return
+
+        try:
+            fn, lno, func, sinfo = logger.findCaller()
+        except ValueError:  # pragma: no cover
+            fn, lno, func = "(unknown file)", 0, "(unknown function)"
+
+        if is_py2:
+            record = logger.makeRecord(
+                logger.name, level, fn, lno, msg, [], None, func, None)
+        else:
+            record = logger.makeRecord(
+                logger.name, level, fn, lno, msg, [], None, func, None, sinfo)
```

Comment:
:-) Well done! I should have concluded this be thinking, too, since the test only requires a variable look-up, while the call requires setting up a stack-frame.